context.go

```diff
@@ -201,13 +208,38 @@ func (context *Context) logFrame(format string, params ...interface{}) {
 	}
 }
 
-func includesProtocol(header string, protocol string) bool {
+// Verify whether the websocket header which is of the format:
+//
+// "WebSocketSubProtocol1, WebSocketSubProtocol2+CBMobile_2"
+//
+// includes at least one protocol with a prefix equivalent to the protocolPrefix passed in.
+// For example, if the protocolPrefix is BLIP_3, then this would be a valid header:
+//
+// "BLIP_3+whatever"
+//
+// Returns a boolean indicating if any of the subprotocols listed in the header string matched.  If so,
+// it returns the matching subprotocol.
+func includesProtocolWithPrefix(header string, protocolPrefix string) (matched bool, matchedSubprotocol string) {
```

Comment:
This isn't quite what I intended for the "+CBMobile_2" part. It refers to the replication protocol (and version), to distinguish it from some other usage of BLIP. That way if a client connects asking for protocol "BLIP_3+MultiPlayerGame" you can reject it because it's not a replication client. Or if a future version of Couchbase Lite sends "BLIP_3+CBMobile_3", you can reject it because you don't know that replication protocol.

In other words, the Context should still compare the protocol string exactly. But the required protocol string needs to be configurable since go-blip shouldn't hardcode the "+CBMobile_2" part.

Reply:
@snej ok makes sense.  I pushed up a new PR, take a look and let me know if that's more along the lines of what you were thinking.